### PR TITLE
19327: Improves train performance by priming of query caches and updating inactive features outside of train flow

### PR DESCRIPTION
**Pull request by howsoRes**

### CODE REVIEW

File: trainee_template/editing.amlg [hunk 1]
```diff
@@ -236,12 +236,16 @@
...
+							inactiveFeaturesMap (remove inactiveFeaturesMap feature)
+							inactiveFeaturesNeedCaching (false)
+						))
 
 						;else removing the only inactive feature, clear out the properties
 						(assign_to_entities (assoc
 							inactiveFeaturesMap (null)
 							hasInactiveFeatures (false)
+							inactiveFeaturesNeedCaching (false)
 						))
```
Comment (howsohazard) [inline trainee_template/editing.amlg]: If you train and then you remove a feature, doesn't this clear inactiveFeaturesNeedCaching when it shouldn't?

File: trainee_template/react.amlg [hunk 2]
```diff
@@ -46,6 +46,8 @@
 			warnings (assoc)
 		))
 
+		(if inactiveFeaturesNeedCaching (call UpdateInactiveFeatures))
```
Comment (howsohazard) [inline trainee_template/react.amlg:49]: Isn't this also needed in  react_series and react_group?  I don't see those files in the PR.

File: trainee_template/editing.amlg (hunk 1 above)
Comment (howsoRes) [reply trainee_template/editing.amlg]: good point. refactored this section

File: trainee_template.amlg [hunk 3]
```diff
@@ -355,14 +354,11 @@
 			;example: (assoc "day" 7 "degrees" 360)
 			cyclicFeaturesMap (null)
 
-			;assoc of feature -> observational_error value as specified by the user
+			;assoc of feature -> observational_error value as specified by the user, set to null when empty
```
Comment (howsohazard) [inline trainee_template.amlg]: did you mean this comment to be on inactiveFeaturesMap below?  or should this be initialized to null?

**Review by howsohazard — APPROVED** [review]

**Pull request merged**